test/integration/test-amp-carousel.js

```diff
@@ -72,19 +79,33 @@ t.run('amp-carousel', function () {
         }
       );
 
-      // TODO(#29783): De-flake and un-skip this test.
-      it.skip(
+      it(
         'should not have any buttons visible when theres only a single ' +
           'item',
-        () => {
+        async () => {
           document.body.classList.add('amp-mode-mouse');
           const amp = document.querySelector('#carousel-1');
           const prevBtn = amp.querySelector('.amp-carousel-button-prev');
           const nextBtn = amp.querySelector('.amp-carousel-button-next');
+
+          async function waitForLayout(el) {
+            const isButtonHidden = () =>
+              el.signals().get(CommonSignals.LOAD_END);
+            await waitFor(isButtonHidden, 'button hidden');
+          }
+
+          waitForLayout(amp);
```

Comment:
each of these `waitForLayout` and `waitForClass` needs to have an `await` before it